purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -127,15 +127,35 @@ def _compute_qty_refunded(self):
         'move_ids.returned_move_ids.state',
     )
     def _compute_qty_returned(self):
+        """Made through read_group for not impacting in performance."""
+        ProductUom = self.env['product.uom']
+        StockMove = self.env['stock.move']
+        groups = StockMove.read_group(
+            [('origin_returned_move_id.purchase_line_id', 'in', self.ids),
+             ('state', '=', 'done'),
+             ('location_id.usage', '!=', 'supplier')],
+            ['origin_returned_move_id', 'product_uom_qty', 'product_uom'],
+            ['origin_returned_move_id', 'product_uom'], lazy=False,
+        )
+        p = self._prefetch
```

Comment:
`p = self._prefetch` is using the same pointer variable.
if you use `id(p)` and `id(self._prefetch)` you will the same id result.

What about uses the same `self` without extra `p` variable and without extra `with_prefetch(p)`?

Reply:
He can use `self._prefetch` directly without `p`, but he needs to add `.with_prefetch()` after a browse call if he wants to reduce SQL queries, because `.browse()` automatically sets a new prefetch object if it doesn't get another one explicitly.